app/src/main/java/org/openobservatory/ooniprobe/activity/MeasurementDetailActivity.java

```diff
@@ -130,20 +129,17 @@ public static Intent newIntent(Context context, int id) {
 		switch (item.getItemId()) {
 			case R.id.rawData:
 				try {
-					FileInputStream is = new FileInputStream(Measurement.getEntryFile(this, measurement.id, measurement.test_name));
-					String json = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create().toJson(new JsonParser().parse(new InputStreamReader(is)).getAsJsonObject());
+					String json = FileUtils.readFileToString(Measurement.getEntryFile(this, measurement.id, measurement.test_name), StandardCharsets.UTF_8);
```

Comment:
In looking at the code behind `getEntryFile` I found it quite surprising that:
```
	static File getMeasurementDir(Context c) {
		File dir = new File(c.getFilesDir(), Measurement.class.getSimpleName());
		dir.mkdirs();
		return dir;
	}
```

is calling `dir.mkdirs` which is a side-effect I would not expect from a getter method.

Moreover according to the android docs: https://developer.android.com/reference/android/content/Context.html#getFilesDir()

> The returned path may change over time if the calling app is moved to an adopted storage device, so only relative paths should be persisted.

Hence repeated calls to `getMeasurementDir` may not yield the same value and create unexpected side effects if not properly controlled.

My suggestion would be to at least explicitly call the directory tree creation function separately from determining the measurement filepath.